internal/provider/resource_apply_module.go

```diff
@@ -260,8 +313,14 @@ func (t *applyModuleResource) Read(ctx context.Context,
 		state.ModuleVersion = types.StringNull()
 	}
 
-	// TODO: Eventually, when the API and SDK support speculative plan runs with a module source,
-	// this should do a speculative plan run here to determine whether changes are needed.
+	// Get the resolved variables from the run that produced the state.
+	resolvedVars, diags := t.toProviderOutputVariables(ctx, currentApplied.resolvedVariables)
```

Comment:
The resolved run variables can't be set in the READ function since they won't be available until after the run has been created. In the CREATE and UPDATE function the variables will need to be queried from the run that was created.